File: Python/07_relay.py
#!/usr/bin/env python

import RPi.GPIO as GPIO
import time


# GPIO0 connect to relay's control pin
# led connect to relay's NormalOpen pin
# 5v connect to relay's com pin
# Set #17 as contral pin
relayPin = 17

# Define a function to print message at the beginning
def print_message():
	print ("========================================")
	print ("|                 Relay                |")
	print ("|    ------------------------------    |")
	print ("| GPIO0 connect to relay's control pin |")
	print ("| led connect to relay's NormalOpen pin|")
	print ("|  5v connect to relay's com pin       |")
	print ("|                                      |")
	print ("|      Make relay to contral a led     |")
	print ("|                                      |")
	print ("|                            SunFounder|")
	print ("========================================\n")
	print 'Program is running...'
	print 'Please press Ctrl+C to end the program...'
	raw_input ("Press Enter to begin\n")

# Define a setup function for some setup
def setup():
	# Set the GPIO modes to BCM Numbering
	GPIO.setmode(GPIO.BCM)
	# Set relayPin's mode to output, 
	# and initial level to High(3.3v)
	GPIO.setup(relayPin, GPIO.OUT, initial=GPIO.HIGH)

# Define a main function for main process
def main():
	# Print messages
	print_message()
	while True:
		print '...Relay close'
		# Tick
		GPIO.output(relayPin, GPIO.LOW)
		time.sleep(1)
		print 'Relay open...'
		# Tock
		GPIO.output(relayPin, GPIO.HIGH) 
		time.sleep(1)

# Define a destroy function for clean up everything after
# the script finished 
def destroy():
	# Turn off LED
	GPIO.output(relayPin, GPIO.HIGH)
	# Release resource
	GPIO.cleanup()                     

# If run this script directly, do:
if __name__ == '__main__':
	setup()
	try:
		main()
	# When 'Ctrl+C' is pressed, the child program 
	# destroy() will be  executed.
	except KeyboardInterrupt:
		destroy()

File: C/07_relay.c
/**********************************************************************
* Filename 		: 4N35.c
* Description 	: Make a relay to contral led blinking
* Author 		: Dream
* E-mail 		: support@sunfounder.com
* Website 		: www.sunfounder.com
* Update 		: Dream    <2016-07-26>
*                       : psvehla  <2019-04-26>
* Detail		: <update details>
*                       : Fixed hi-lo mixup.
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>

#define RelayPin 0

int main(void){
	if(wiringPiSetup() == -1){ //when initialize wiring failed, print messageto screen
		printf("setup wiringPi failed !");
		return 1; 
	}
	
	pinMode(RelayPin, OUTPUT);   //set GPIO0 output

	printf("\n");
	printf("\n");
	printf("========================================\n");
	printf("|                 Relay                |\n");
	printf("|    ------------------------------    |\n");
	printf("| GPIO0 connect to relay's control pin |\n");
	printf("| led connect to relay's NormalOpen pin|\n");
	printf("|  5v connect to relay's com pin       |\n");
	printf("|                                      |\n");
	printf("|      Make relay to contral a led     |\n");
	printf("|                                      |\n");
	printf("|                            SunFounder|\n");
	printf("========================================\n");
	printf("\n");
	printf("\n");

	while(1){
		// Tick
		printf("......Relay Close\n");
		digitalWrite(RelayPin, HIGH);
		delay(1000);
		// Tock
		printf("Relay Open......\n");
		digitalWrite(RelayPin, LOW);
		delay(1000);
	}

	return 0;
}


